### PR TITLE
Ruby Project #4-2: Enumerables tests

### PR DESCRIPTION
**Pull request by mateomh**
### Enumerable testing with RSPEC

We created the tests for the code done in the enumerable methods project with **rspec**. It has both good and bad scenarios for the tests with a total of 46 tests.

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,256 @@
+require './lib/enumerable.rb'
+
+describe Enumerable do
+  include Enumerable
+
+  describe 'My Each method' do
+    it 'Takes each element of the object passed' do
+      expect([1, 2, 5, 6, 8].my_each { |x| print x }).to eql([1, 2, 5, 6, 8].each { |x| print x })
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: - Lets test for when no block is given

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [x] All methods implemented in the Enumerables project should be tested ,that should include:
- my_each_with_index
- my_select
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/all_enumerables.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/testing-your-ruby-code))
- [x] Specs for each method should be wrapped in one `describe` block. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/describe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

- [x] Check comments below



### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [x] Let's write tests for the following methods when a block is given .

- my_each
- my_each_with_index
- my_inject
- my_count
- my_any
- my_all
- my_select
- my_none


Note: I apologise that this wasn't mentioned in my previous review.
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyoa test

**Review by mikenath223 — COMMENTED** [review]
### Hi Team 👋


Nice work so far on this.


- [x] There are some tests that check empty arrays. Kindly add tests for an array filled with elements.
- [x] There are conflicts with the default branch on these files: `.github/workflows/linters.yml` `lib/enumerable.rb` `spec/spec_helper.rb`. Please fix them.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Pull request merged**